core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -154,6 +155,33 @@ static HttpResponse delayed(HttpResponse response, Duration delay, ScheduledExec
         return res;
     }
 
+    /***
+     * Invokes the specified {@link Supplier} and creates a new HTTP response that
+     * delegates to the provided {@link HttpResponse} by {@link Supplier},
+     * beginning publishing after {@code delay} has passed from a random {@link ScheduledExecutorService}.
+     */
+    static HttpResponse delayed(Supplier<? extends HttpResponse> responseSupplier, Duration delay) {
+        requireNonNull(responseSupplier, "responseSupplier");
+        requireNonNull(delay, "delay");
+        return delayed(responseSupplier, delay, CommonPools.workerGroup().next());
```

Comment:
Should we use the current thread-local `RequestContext`'s event loop if possible?

Reply:
Then is it okay to replace `CommonPools.workerGroup().next()` to `RequestContext.current().eventLoop()`? What is the meaning of the if possible?.?

Reply:
You may want to see https://github.com/line/armeria/blob/3e2185083fdd7ad45b9b7a71cf5b573cbd8791d2/core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java#L442

Reply:
Fully what I want! Thank you 🙂